grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,372 @@
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].storage_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["storages_id"]).all()
+            
+    def check_orig_ids(self, ma_env, domain : dict, space = 'action'):
+        #It tests if the origin ids are correct
+        
+        assert (np.sort(np.concatenate([
+            ma_env._subgrids_cls[space][agent].sub_orig_ids 
+            for agent in domain.keys()])) == np.arange(ma_env._cent_env.n_sub)).all()
+        
+        for agent in domain.keys():
+            mask_load = ma_env._subgrids_cls[space][agent].mask_load
+            mask_gen = ma_env._subgrids_cls[space][agent].mask_gen
+            mask_storage = ma_env._subgrids_cls[space][agent].mask_storage
+            mask_line_or = ma_env._subgrids_cls[space][agent].mask_line_or
+            mask_line_ex = ma_env._subgrids_cls[space][agent].mask_line_ex
+            mask_shunt = ma_env._subgrids_cls[space][agent].mask_shunt
+            mask_interco = ma_env._subgrids_cls[space][agent].mask_interco
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_orig_ids == np.sort(domain[agent])).all()
+            assert (ma_env._subgrids_cls[space][agent].gen_orig_ids\
+                ==\
+                    np.arange(ma_env._cent_env.n_gen)[mask_gen]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].load_orig_ids\
+                ==\
+                    np.arange(ma_env._cent_env.n_load)[mask_load]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].storage_orig_ids\
+                ==\
+                    np.arange(ma_env._cent_env.n_storage)[mask_storage]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].line_orig_ids\
+                ==\
+                    np.arange(ma_env._cent_env.n_line)[mask_line_or]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].line_orig_ids\
+                ==\
+                    np.arange(ma_env._cent_env.n_line)[mask_line_ex]).all()
+            
+            if ma_env._subgrids_cls[space][agent].n_shunt > 0:
+                assert (ma_env._subgrids_cls[space][agent].shunt_orig_ids\
+                    ==\
+                        np.arange(ma_env._cent_env.n_shunt)[mask_shunt]).all()
+                
+            assert (ma_env._subgrids_cls[space][agent].interco_to_lineid\
+                ==\
+                    np.arange(ma_env._cent_env.n_line)[mask_interco]).all()
+            
+            
+    def check_connections(self, ma_env, domain, space = 'action'):
+        for agent in domain.keys():
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_info\
+                ==\
+                    ma_env._cent_env.sub_info[
+                        ma_env._subgrids_cls[space][agent].sub_orig_ids
+                    ]).all()
```

Comment:
Not enough. Please plase please take some notes !

We want also to check that "if two objects are on the same substations on the original grid, they are also on the same substation on the subgrid"
and the opposite

"if two objects are on the same substations on the subgrid, they are on the same substation on the original grid"